Modules/Core/Common/include/itkThreadedImageRegionPartitioner.h

```diff
@@ -60,7 +60,7 @@ class ITK_TEMPLATE_EXPORT ThreadedImageRegionPartitioner : public ThreadedDomain
   itkTypeMacro(ThreadedImageRegionPartitioner, ThreadedDomainPartitioner);
 
   /** Type of the object being threaded over */
-  using DomainType = typename Superclass::DomainType;
+  using typename Superclass::DomainType;
```

Comment:
@hjmjohnson looks like the change here is causing problems on Windows when trying to define the `SizeType` and `IndexType` aliases a few lines below:
https://open.cdash.org/viewBuildError.php?buildid=7451857

Any guess?

I've also noticed that `using ImageRegionType = typename Self::DomainType;` defines yet another alias for the `DomainType`. Not sure both aliases meaning the same type are required.

Reply:
@jhlegarreta I tried to build it locally on Windows, and got the same compilation errors, from VS2019. It appears that it can be solved by doing:

```
  using SizeType = typename ImageRegionType::SizeType;
  using IndexType = typename ImageRegionType::IndexType;

```

Instead of:
https://github.com/InsightSoftwareConsortium/ITK/blob/201537c60b2184b3febf5706d39dce71024cd769/Modules/Core/Common/include/itkThreadedImageRegionPartitioner.h#L68-L69

Reply:
Thanks for having given this a try @N-Dekker :100: . I still wonder why two aliases (`DomainType` and `ImageRegionType`) are created to refer to the same type.

Reply:
Went ahead and adopted the suggestion @N-Dekker proposed. Push-forced in 431d91c. I also realized that there is a comment saying that these types are deprecated (although no pre-processor guard is added). It is left for a separate exercise if anybody has the bandwidth to have a look